### ISSUE TITLE
[unbranded Client CodeGen] Support paging for unbranded

### ISSUE THREAD

**Issue by lirenhe** [microsoft/typespec#5204]
### Clear and concise description of the problem

Based on the current agreement, with `@list`, `@pageItems` and `@nextLink`, we could generate the paging operation similar with Azure style.

The current priority is to solve the problem we are facing for TypeSpec-ToDo to unblock the E2E demo.

```[tasklist]
### actions
- [ ] https://github.com/Azure/autorest.python/issues/2936
- [ ] https://github.com/Azure/autorest.java/issues/2992
- [ ] https://github.com/Azure/autorest.typescript/issues/2932
- [ ] https://github.com/microsoft/typespec/issues/5356
```

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [docs](https://typespec.io/docs/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by ArcturusZhang** [comment]
The unbranded paging core types have been added into System.ClientModel code (a few days ago), but I am not sure if there is a version released for this change.

**Comment by lirenhe** [comment]
Close it as all subitem are resolved.  .NET has a separate issue to track this effort.
